### PR TITLE
[MINOR][DOCS] Mention other Python dependency tools in documentation

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

Self-contained.

### Why are the changes needed?

For user's more information on available Python dependency management in PySpark.

### Does this PR introduce _any_ user-facing change?
Yes, documentation change.

### How was this patch tested?
Manaully built the docs and checked the results:
Screen Shot 2021-09-29 at 10 11 56 AM
Screen Shot 2021-09-29 at 10 12 22 AM
Screen Shot 2021-09-29 at 10 12 42 AM

### CODE REVIEW

File: docs/rdd-programming-guide.md [hunk 1]
```diff
@@ -241,12 +241,12 @@ For a complete list of options, run `spark-shell --help`. Behind the scenes,
 In the PySpark shell, a special interpreter-aware SparkContext is already created for you, in the
 variable called `sc`. Making your own SparkContext will not work. You can set which master the
 context connects to using the `--master` argument, and you can add Python .zip, .egg or .py files
-to the runtime path by passing a comma-separated list to `--py-files`. You can also add dependencies
+to the runtime path by passing a comma-separated list to `--py-files`. For thrid-party Python dependencies,
```
Comment (mkgray) [inline docs/rdd-programming-guide.md:244]: Minor typo, should be third-party

File: docs/submitting-applications.md [hunk 2]
```diff
@@ -35,7 +35,8 @@ script as shown here while passing your jar.
 
 For Python, you can use the `--py-files` argument of `spark-submit` to add `.py`, `.zip` or `.egg`
 files to be distributed with your application. If you depend on multiple Python files we recommend
-packaging them into a `.zip` or `.egg`.
+packaging them into a `.zip` or `.egg`. For thrid-party Python dependencies,
```
Comment (mkgray) [inline docs/submitting-applications.md:38]: Minor typo, should be third-party

**Review by mkgray — CHANGES_REQUESTED** [review]

File: docs/rdd-programming-guide.md (hunk 1 above)
Comment (HyukjinKwon) [reply docs/rdd-programming-guide.md:244]: oops, thanks man

Comment (HyukjinKwon) [inline docs/rdd-programming-guide.md:244]:
```suggestion
to the runtime path by passing a comma-separated list to `--py-files`. For third-party Python dependencies,
```

File: docs/submitting-applications.md (hunk 2 above)
Comment (HyukjinKwon) [inline docs/submitting-applications.md:38]:
```suggestion
packaging them into a `.zip` or `.egg`. For third-party Python dependencies,
```

**Review by viirya — APPROVED** [review]

**Review by ueshin — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Thanks guys.

Merged to master. Let me merge this in branch-3.2 too - very trivial doc change.

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.2.

**Comment by dongjoon-hyun** [comment]
+1, late LGTM.

**Pull request closed without merging**